components/Editor.js

```diff
@@ -975,6 +989,20 @@ var _self =
             (n.tokens = C.tokenize(n.code, n.grammar)),
             C.hooks.run('after-tokenize', n),
             M.stringify(C.util.encode(n.tokens), n.language)
+              .split(/\n/)
+              .map((x, i) => {
+                const line = i + 1;
+                return `<span id="line-${line}" class="line-number${
+                  line == getSafeHash() ? ' selected' : ''
+                }" onclick="const clicked = this.id.replace('line-', '');
```

Comment:
Instead of generating this callback function for every row, could we add this to the parent element and rely on event bubbling to catch it?

Reply:
Yep I think this works better! The reason I didn't go with that initially is that I couldn't get the event to bubble reliably, I think with the way I'm styling with negative margin etc.

Making it work with bubbling would make doing line ranges a bit easier (or at least nicer-looking) so maybe I'll WIP this again to address the new bugs and have a go at including line ranges in MVP